Description: XSS detection code is now implememted with internal logic and it
 does no longer depend on the external detectxsslib library (not GPL-compatible).
Origin: https://github.com/ntop/ntopng/commit/badcf525cb47efe74d577878a849c71141b80621

Index: ntopng/src/Utils.cpp
===================================================================
--- ntopng.orig/src/Utils.cpp
+++ ntopng/src/Utils.cpp
@@ -24,11 +24,6 @@
 #include <curl/curl.h>
 #include <string.h>
 
-extern "C" {
-#include "../third-party/detectxsslib/detectxsslib.h"
-#include "../third-party/detectxsslib/detectxsslib.c"
-};
-
 // A simple struct for strings.
 typedef struct {
   char *s;
@@ -629,20 +624,46 @@ bool Utils::isUserAdministrator(lua_Stat
  * @param param   The parameter to purify (remove unliked chars with _)
  */
 
-void Utils::purifyHTTPparam(char *param, bool strict) {
-  xsslibUrl url;
-
-  /* Fix for http://packetstormsecurity.com/files/127329/Ntop-NG-1.1-Cross-Site-Scripting.html */
-  xsslibUrlInit(&url);
-  xsslibUrlSetUrl(&url, param);
+static const char* xssAttempts[] = {
+  "<?import",
+  "<applet",
+  "<base",
+  "<embed",
+  "<frame",
+  "<iframe",
+  "<implementation",
+  "<import",
+  "<link",
+  "<meta",
+  "<object",
+  "<script",
+  "<style",
+  "charset",
+  "classid",
+  "code",
+  "codetype",
+  "data",
+  "href",
+  "http-equiv",
+  "javascript:",
+  "src",
+  "type",
+  "vbscript:",
+  "vmlframe",
+  "xlink:href",
+  "=",
+  NULL
+};
 
-  if(xsslibUrlScan(&url) != XssClean) {
-    ntop->getTrace()->traceEvent(TRACE_WARNING, "Found possible XSS attempt: %s", param);
-    param[0] = '\0';
+void Utils::purifyHTTPparam(char *param, bool strict) {
+  for(int i=0; xssAttempts[i] != NULL; i++) {
+    if(strstr(param, xssAttempts[i])) {
+      ntop->getTrace()->traceEvent(TRACE_WARNING, "Found possible XSS attempt: %s", param);
+      param[0] = '\0';
+      return;
+    }
   }
 
-
-#if 0
   for(int i=0; param[i] != '\0'; i++) {
 
     bool is_good;
@@ -680,7 +701,6 @@ void Utils::purifyHTTPparam(char *param,
       param[i-1] = '_', param[i] = '_'; /* Invalidate the path */
     }
   }
-#endif
 }
 
 /* **************************************************** */
